### PR TITLE
Refactor scicat dataset instance creation.

### PR DESCRIPTION
**Pull request by YooSunYoung**
Use ``ScicatDataset`` dataclass to create a scicat dataset entry instead of plain ``dict``.

### CODE REVIEW

File: src/scicat_dataset.py [hunk 1]
```diff
@@ -54,6 +65,45 @@ def convert_to_type(input_value: Any, dtype_desc: str) -> Any:
...
+            raise Exception("Invalid variable source: ", source)
+        variable_map[variable_name] = convert_to_type(
+            value, variable_recipe["value_type"]
+        )
+    return variable_map
+
+
 @dataclass(kw_only=True)
 class TechniqueDesc:
     pid: str
```
Comment (nitrosx) [inline src/scicat_dataset.py:109]: Technique has an additional field named "name". Its type is string

File: src/scicat_communication.py
Comment (nitrosx) [inline src/scicat_communication.py:1]: Should we transfer in this file all the communication with SciCat?

**Review by nitrosx — CHANGES_REQUESTED** [review]

File: src/scicat_dataset.py (hunk 1 above)
Comment (nitrosx) [reply src/scicat_dataset.py:109]: One more thing: techniques can be empty.

Comment (YooSunYoung) [reply src/scicat_dataset.py:109]: > Technique has an additional field named "name". Its type is string

@nitrosx  is it `name` or `names`...? 
If you expand the changes, there is a field called `names`. I copied it from the previous implementation.

> One more thing: techniques can be empty.

Yes, it's handled here:
https://github.com/SciCatProject/scicat-filewriter-ingest/blob/3f3359e79ffeeb2e19e50415cc8f51f7a61263ab/src/scicat_dataset.py#L467-L469

Or you mean not-setting it shouldn't be the same as setting an empty list...?

File: src/scicat_communication.py
Comment (YooSunYoung) [reply src/scicat_communication.py:1]: Yes that's what #62 about.

File: src/scicat_dataset.py (hunk 1 above)
Comment (nitrosx) [reply src/scicat_dataset.py:109]: Dataset techniques fields is a list of dictionaries with the following structure:
```
{
 'pid' : str,
 'name' : str
}
```

If no technique is provided, the field should be an empty array.
Right now, I'm planning to leave the field empty. Future upgrades will include them. I'm still talking to the instrument teams to figure out which information they would like to have in here

File: src/scicat_communication.py
Comment (nitrosx) [reply src/scicat_communication.py:1]: Great. Thanks

**Review by nitrosx — APPROVED** [review]

**Pull request merged**